kevm-pyk/src/kevm_pyk/interpreter.py

```diff
@@ -15,14 +16,30 @@
 
 
 def iterate_gst(
-    gst_data: dict, schedule: str, mode: str, chainid: int, usegas: bool, skipped_keys: frozenset[str] = frozenset()
+    gst_data: dict,
+    mode: str,
+    chainid: int,
+    usegas: bool,
+    skipped_keys: frozenset[str] = frozenset(),
+    schedule: str | None = None,
 ) -> Iterator[tuple[str, App]]:
     """Yield (test_name, kore_pattern) for each test in GST data after filtering discarded keys."""
     for test_name, test in gst_data.items():
         if test_name in skipped_keys:
             continue
+        test_schedule = _resolve_schedule(test, test_name, schedule)
         gst_filtered = {test_name: filter_gst_keys(test)}
-        yield test_name, gst_to_kore(gst_filtered, schedule, mode, chainid, usegas)
+        yield test_name, gst_to_kore(gst_filtered, test_schedule, mode, chainid, usegas)
+
+
+def _resolve_schedule(test: dict, test_name: str, fallback: str | None) -> str:
+    """Return schedule from test's network field, falling back to provided schedule or DEFAULT_SCHEDULE."""
+    network = test.get('network')
+    if network is not None:
+        if network not in SCHEDULE_MAPPING:
+            raise ValueError(f'Unknown network {network!r} in test {test_name!r}')
+        return SCHEDULE_MAPPING[network]
+    return fallback or DEFAULT_SCHEDULE
```

Comment:
I'm wondering why we allow passing in the schedule at all to `iterate_gst`, because shouldn't every GST have a schedule defined already in it?

If they don't have the schedule already defined, is the idea that VMTests get `DEFAULT` scehduel by default, and BlockchainTests get `PRAGUE` by default?

Are there any vmtests or bchain tests that don't have the schedule defined in them?